astropy/coordinates/sky_coordinate.py

```diff
@@ -900,9 +903,13 @@ def match_to_catalog_3d(self, catalogcoord, nthneighbor=1):
         """
         from .matching import match_coordinates_3d
 
+
         if (isinstance(catalogcoord, (SkyCoord, BaseCoordinateFrame))
                 and catalogcoord.has_data):
-            self_in_catalog_frame = self.transform_to(catalogcoord)
+            if self.is_equivalent_frame(catalogcoord):
+                self_in_catalog_frame = self
+            else:
+                self_in_catalog_frame = self.transform_to(catalogcoord)
```

Comment:
Hmm, yes, and then presumably also for `match_to_catalog_sky` (which was not changed in #5723)